folding-schemes/src/folding/circuits/cyclefold.rs

```diff
@@ -520,13 +512,19 @@ impl<CFG: CycleFoldConfig, GC: CurveVar<CFG::C, CF2<CFG::C>>> ConstraintSynthesi
             .flatten()
             .collect();
 
-        let computed_x = [
+        let x = [
             r_fp,
             points_aux,
             p_folded.to_constraint_field()?[..2].to_vec(),
         ]
         .concat();
-        computed_x.enforce_equal(&x)?;
+        #[cfg(test)]
+        assert_eq!(x.len(), CFG::IO_LEN); // non-constrained sanity check
+
+        // This line "converts" `x` from a witness to a public input.
+        // Instead of directly modifying the constraint system, we explicitly
+        // allocate a public input and enforce that its value is indeed `x`.
+        Vec::new_input(cs.clone(), || x.value())?.enforce_equal(&x)?;
```

Comment:
This (and other similar lines) may look weird because `x` is compared with itself. In fact, `x.value()` here allows an honest prover to extract the public inputs, and `enforce_equal` prevents a malicious prover from claiming incorrect public inputs that are different from the honest `x` computed by the circuit.

Reply:
Thanks for the comment! maybe this comment could be added to the inline code comment? (the reasoning about the honest prover & malicious prover).
(Same for the other occurrences of this line in `{hypernova, nova, protogalaxy}/circuits.rs`)